### ISSUE TITLE
[Firefox] Extension signing

### ISSUE THREAD

**Issue by Dimas-sc** [uBlock-LLC/uBlock#764]
https://blog.mozilla.org/addons/2015/02/10/extension-signing-safer-experience/

**Comment by ghost** [comment]
#747

**Comment by gorhill** [comment]
@hotdogman 

> Extensions that are submitted for hosting on AMO and pass review will be automatically signed.

uBlock will be on AMO.

**Comment by ghost** [comment]
> uBlock will be on AMO.

Soon™

**Comment by SammysHP** [comment]
The main issue is that you need a developer build of Firefox to install the add-on without sending it to Mozilla first. This is a huge drawback because I compile it myself with some small patches, but don't want to use a special build of Firefox. And it is annoying to compile the xpi first, then upload it via a web interface, wait, download it again etc.

I hope they'll implement some sort of local signing with own certificate that can be added to your Firefox installation.

**Comment by xeoron** [comment]
There might be a way to disable the signing check in about:about, once FF releases this new feature. 

With that said, even if uBlock only submits the even or odd number version updates for Mozilla to sign, it should not be that hard for them to do. The bigger question is: why is it not in the AMO listing already?

**Comment by ghost** [comment]
> The bigger question is: why is it not in the AMO listing already?

@xeoron 

https://github.com/gorhill/uBlock/issues/550#issuecomment-72313569

tl;dr; Some anon uploaded the file to AMO, blocking the file from subsequent upload, setting everything back by a few weeks.

**Comment by ghost** [comment]
@xeoron commented on [11 févr. 2015 17:49 UTC+1](https://github.com/gorhill/uBlock/issues/764#issuecomment-73916300):

> There might be a way to disable the signing check in about:about, once FF releases this new feature. 

Are we even sure Mozilla will release this new extension signing feature, or will release it without any amendments? Looks like many developers as well as users have already started complaining, and are doing so with arguments that could defeat a Mozilla's stubbornness should the company be tightly focused on their aim.

**Comment by SohKa** [comment]
> There might be a way to disable the signing check in about:about, once FF releases this new feature.

Right now, this is not their intention.

The complete sentence, from the article (https://blog.mozilla.org/addons/2015/02/10/extension-signing-safer-experience/) :

> After the transition period, it will not be possible to install unsigned extensions in Release or Beta versions of Firefox. There won’t be any preferences or command line options to disable this.

**Comment by lewisje** [comment]
It looks like Deathamns is the official steward of the Firefox build, considering this link from this project's README: https://addons.mozilla.org/en-US/firefox/addon/ublock/
